File: platform-sdk-java/platform-sdk-process/src/main/java/com/platform/sdk/process/ProcessStartBuilder.java
package com.platform.sdk.process;

import com.fasterxml.jackson.core.type.TypeReference;
import com.platform.sdk.core.http.PlatformHttpClient;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class ProcessStartBuilder {

    private final PlatformHttpClient http;
    private final String processKey;
    private String businessKey;
    private String idempotencyKey;
    private String callbackUrl;
    private final Map<String, Object> variables = new HashMap<>();

    ProcessStartBuilder(PlatformHttpClient http, String processKey) {
        this.http = http;
        this.processKey = processKey;
    }

    public ProcessStartBuilder businessKey(String businessKey) {
        this.businessKey = businessKey;
        return this;
    }

    public ProcessStartBuilder idempotencyKey(String idempotencyKey) {
        this.idempotencyKey = idempotencyKey;
        return this;
    }

    public ProcessStartBuilder variable(String name, Object value) {
        this.variables.put(name, value);
        return this;
    }

    public ProcessStartBuilder variables(Map<String, Object> vars) {
        this.variables.putAll(vars);
        return this;
    }

    public ProcessStartBuilder callbackUrl(String callbackUrl) {
        this.callbackUrl = callbackUrl;
        return this;
    }

    public ProcessTracker submit() throws IOException {
        Map<String, Object> body = new HashMap<>();
        body.put("processKey", processKey);
        if (businessKey != null) body.put("businessKey", businessKey);
        if (idempotencyKey != null) body.put("idempotencyKey", idempotencyKey);
        if (callbackUrl != null) body.put("callbackUrl", callbackUrl);
        if (!variables.isEmpty()) body.put("variables", variables);
        return http.post("/v1/processes", body, new TypeReference<>() {});
    }
}

File: platform-sdk-js/src/process/ProcessStartBuilder.ts
import { PlatformClient } from '../client/PlatformClient'
import { ProcessTracker } from './types'

export class ProcessStartBuilder {
  private _businessKey?: string
  private _idempotencyKey?: string
  private _callbackUrl?: string
  private readonly _variables: Record<string, unknown> = {}

  constructor(
    private readonly client: PlatformClient,
    private readonly processKey: string,
  ) {}

  businessKey(bk: string): this { this._businessKey = bk; return this }
  idempotencyKey(ik: string): this { this._idempotencyKey = ik; return this }
  callbackUrl(url: string): this { this._callbackUrl = url; return this }
  variable(name: string, value: unknown): this { this._variables[name] = value; return this }
  variables(vars: Record<string, unknown>): this { Object.assign(this._variables, vars); return this }

  async submit(): Promise<ProcessTracker> {
    const body: Record<string, unknown> = { processKey: this.processKey }
    if (this._businessKey) body.businessKey = this._businessKey
    if (this._idempotencyKey) body.idempotencyKey = this._idempotencyKey
    if (this._callbackUrl) body.callbackUrl = this._callbackUrl
    if (Object.keys(this._variables).length > 0) body.variables = this._variables
    return this.client.post<ProcessTracker>('/v1/processes', body)
  }
}
